plaso/storage/sqlite/sqlite_file.py

```diff
@@ -194,6 +197,10 @@ def _CheckStorageMetadata(cls, metadata_values):
     except (TypeError, ValueError):
       raise IOError('Invalid format version: {0!s}.'.format(format_version))
 
+    if not check_readable_only and format_version != cls._FORMAT_VERSION:
```

Comment:
why not use something like `check_writable` and reverse the logic?

Reply:
This was to match the "read_only" behavior in the storage interface.